### PR TITLE
Add method to find API key in environment or key file

### PR DESCRIPTION
**Pull request by perlinm**
Porting [`css#456`](https://github.com/SupertechLabs/cirq-superstaq/pull/456), per [suggestion](https://github.com/SupertechLabs/cirq-superstaq/pull/456#discussion_r1134717267).

Add a method to find a SuperstaQ API key from the environment or a key file.

### CODE REVIEW

File: general_superstaq/superstaq_client.py [hunk 1]
```diff
@@ -417,3 +419,25 @@ def __repr__(self) -> str:
                 verbose={self.verbose!r},
             )"""
         )
+
+
+def _find_api_key() -> Optional[str]:
+    """Try to load a SuperstaQ API key from the environment or a key file."""
+
+    if api_key := os.getenv("SUPERSTAQ_API_KEY"):
```
Comment (perlinm) [inline general_superstaq/superstaq_client.py:427]: Walrus operator not allowed for [another 3 months](https://devguide.python.org/versions/) (at which point python3.7 gets retired) :(

**Comment by stephanielee9** [comment]
Update dev guide

**Review by bharat-thotakura — APPROVED** [review]
LGTM

**Comment by richrines1** [comment]
@bharat-thotakura thanks for reviewing! this pr was moved to the public monorepo: https://github.com/SupertechLabs/client-superstaq/pull/408

this repo along with https://github.com/SupertechLabs/cirq-superstaq/ and https://github.com/SupertechLabs/qiskit-superstaq/pulls are now inactive - all of our public code is now in https://github.com/SupertechLabs/client-superstaq. i'll go through and move/close the lingering prs in the other repos now - no need to review them until they've been reopened in the monorepo

**Comment by richrines1** [comment]
@perlinm closing this, as it was done in https://github.com/SupertechLabs/client-superstaq/pull/408

**Pull request closed without merging**